File: moviejava/app/src/main/java/com/example/moviejava/data/remote/people/PeopleService.java
package com.example.moviejava.data.remote.people;


import com.example.moviejava.model.JsonObject;
import com.example.moviejava.model.Person;

import io.reactivex.rxjava3.core.Observable;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface PeopleService {
    @GET("person/{person_id}")
    Observable<Person> getPerson(@Path("person_id") int id);

    @GET("discover/movie")
    Observable<JsonObject> getMovieByPerson(@Query("with_people") int id);
}

File: movie/app/src/main/java/com/example/movie/data/remote/people/PeopleService.kt
package com.example.movie.data.remote.people

import com.example.movie.data.remote.Constant
import com.example.movie.model.film.Film
import com.example.movie.model.JsonObject
import com.example.movie.model.Person
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface PeopleService {
    @GET("person/{person_id}")
    suspend fun getPerson(@Path("person_id") id: Int): Person

    @GET("discover/movie")
    suspend fun getMovieByPerson(@Query("with_people") id: Int): JsonObject
}